src/quest/historian.py

```diff
@@ -845,7 +841,7 @@ async def _run_with_args(self, *args, **kwargs):
         kwargs = await self.handle_step('kwargs', lambda: kwargs)
         result = await self.handle_step(get_function_name(self.workflow), self.workflow, *args, **kwargs)
         self._workflow_completed = True
-        await self._handle_resource_update(None)
+        self._resource_stream_manager.notify_of_workflow_stop()
```

Comment:
See new method notify_of_workflow_stop in resources.py. Note that the difference here is that by calling handle resource update, the completing of the workflow is treated like a resource update and the generator will yield get_resources. With the new method, nothing is yielded and the generator simply exits.

Reply:
Good question. I think we want workflow completion to be an event, but suspend can simply end the resource stream loop.